src/main/java/www/purple/mixxy/models/User.java

```diff
@@ -26,6 +26,24 @@
 	// @Chris - TODO: how are we going to use this map?
 	// public Map<OAuthService, String> authentications = new HashMap<>();
 
+	
+	//TODO: Partial Indexing of MOD roles
+	// https://github.com/objectify/objectify/wiki/Queries#custom-conditions
+	
+//	static class IfMod extends ValueIf<User> {
+//	    public boolean matches(User user) {
+//	        return user.role == Role.MODERATOR;
+//	    }
+//
+//		@Override
+//		public boolean matchesValue(User arg0) {
+//			// TODO Auto-generated method stub
+//			return false;
+//		}
+//	}
+	
+	//TODO: Partial Indexing of MOD roles
```

Comment:
Uncommenting the code and adding the ValueIf import, will allow this to compile cleanly and should work as intended based on the objectify Partial Indexing Doc. When I run it, I receive this error. 

> Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.559 sec <<< FAILURE! - in www.purple.mixxy.controllers.UserTest
> testCreateAndRetrieveUser(www.purple.mixxy.controllers.UserTest)  Time elapsed: 0.438 sec  <<< ERROR!
> java.lang.ExceptionInInitializerError: null
> 	at www.purple.mixxy.controllers.NinjaAppengineBackendTest.setUp(NinjaAppengineBackendTest.java:25)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:606)
> 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
> 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
> 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
> 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
> 	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
> 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
> 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
> 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
> 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
> 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
> 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
> 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
> 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
> 	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
> 	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:264)
> 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
> 	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
> 	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
> 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
> 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
> Caused by: java.lang.IllegalStateException: Cannot use If class www.purple.mixxy.models.User$IfMod on public www.purple.mixxy.models.Role www.purple.mixxy.models.User.role because you cannot assign www.purple.mixxy.models.Role to www.purple.mixxy.models.User
> 	at com.googlecode.objectify.impl.IfConditionGenerator.generateIfConditions(IfConditionGenerator.java:53)
> 	at com.googlecode.objectify.impl.FieldProperty.<init>(FieldProperty.java:58)
> 	at com.googlecode.objectify.impl.translate.ClassPopulator.getDeclaredProperties(ClassPopulator.java:209)
> 	at com.googlecode.objectify.impl.translate.ClassPopulator.<init>(ClassPopulator.java:84)
> 	at com.googlecode.objectify.impl.translate.ClassTranslatorFactory.createEntityClassTranslator(ClassTranslatorFactory.java:66)
> 	at com.googlecode.objectify.impl.translate.ClassTranslatorFactory.create(ClassTranslatorFactory.java:49)
> 	at com.googlecode.objectify.impl.translate.ClassTranslatorFactory.create(ClassTranslatorFactory.java:36)
> 	at com.googlecode.objectify.impl.translate.Translators.create(Translators.java:138)
> 	at com.googlecode.objectify.impl.translate.Translators.get(Translators.java:117)
> 	at com.googlecode.objectify.impl.translate.Translators.getRoot(Translators.java:129)
> 	at com.googlecode.objectify.impl.EntityMetadata.<init>(EntityMetadata.java:49)
> 	at com.googlecode.objectify.impl.Registrar.register(Registrar.java:70)
> 	at com.googlecode.objectify.ObjectifyFactory.register(ObjectifyFactory.java:185)
> 	at com.googlecode.objectify.ObjectifyService.register(ObjectifyService.java:64)
> 	at www.purple.mixxy.conf.ObjectifyProvider.<clinit>(ObjectifyProvider.java:38)
> 	at www.purple.mixxy.controllers.NinjaAppengineBackendTest.setUp(NinjaAppengineBackendTest.java:25)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:606)
> 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
> 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
> 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
> 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
> 	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
> 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
> 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
> 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
> 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
> 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
> 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
> 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
> 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
> 	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
> 	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:264)
> 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
> 	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
> 	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
> 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
> 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
> 
> 
> Results :
> 
> Tests in error: 
>   PostTest>NinjaAppengineBackendTest.setUp:25 » ExceptionInInitializer
>   UserTest>NinjaAppengineBackendTest.setUp:25 » ExceptionInInitializer
> 
> Tests run: 17, Failures: 0, Errors: 2, Skipped: 0
> 
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD FAILURE
> [INFO] ------------------------------------------------------------------------
> [INFO] Total time: 15.091 s
> [INFO] Finished at: 2016-04-06T21:59:18-04:00
> [INFO] Final Memory: 28M/175M
> [INFO] ------------------------------------------------------------------------
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.15:test (default-test) on project mixxy: There are test failures.
> [ERROR] 
> [ERROR] Please refer to /Users/Brian_Sabz/Desktop/CSE 308/Final Project/Mixxy/target/surefire-reports for the individual test results.
> [ERROR] -> [Help 1]
> [ERROR] 
> [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
> [ERROR] Re-run Maven using the -X switch to enable full debug logging.
> [ERROR] 
> [ERROR] For more information about the errors and possible solutions, please read the following articles:
> [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
> `